src/interpreter/Environment.ts

```diff
@@ -174,11 +190,30 @@ export class Environment {
         newEnvironment.global = this.global;
         newEnvironment.module = this.module;
         newEnvironment.mPointer = this.mPointer;
+        newEnvironment.mock = this.mock;
+        newEnvironment.mockObjects = this.mockObjects;
         newEnvironment.focusedNode = this.focusedNode;
 
         return newEnvironment;
     }
 
+    /**
+     * retrieves mocked object if it exists
+     * @param objName the object to mock
+     */
+    public getMock(objName: string): BrsType {
```

Comment:
Any particular reason to store this in `Environment`?  I suspect we could use a similar `Map<string, RoAssociativeArray>` stored in `src/mocks/RegisteredMocks.ts` or something, and the `CreateObject` implementation just grabs it from `(up a billion levels)/mocks/RegisteredMocks`

Reply:
Hmm, interestingly my first at doing what I suggested didn't work sooooo 🤷‍♂ this works!